statsmodels/discrete/discrete_model.py

```diff
@@ -4052,12 +4053,23 @@ def __init__(self, model, mlefit):
 
     def _maybe_convert_ynames_int(self, ynames):
         # see if they're integers
-        try:
-            for i in ynames:
+        issue_warning = False
+        msg = 'endog contains values are that not int-like. Uses string ' \
+              'representation of value.\nUse integer-valued endog to ' \
+              'suppress this warning.'
```

Comment:
I'd suggest a) not putting the newline in there and letting the user's terminal handle it and b) wrapping the whole thing in parens instead of ending lines with backslashes.  The latter is slightly because of personal preference, mostly because this finding this backslash usage is a decent heuristic for identifying code in need of attention.